### PR TITLE
Fix raw deployment service points to predictor container port instead of transformer container port in transformer collocation

### LINKED ISSUE #3317: Raw deployment service points to predictor container port instead of transformer container port in transformer collocation
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
All the request are directed to predictor container instead of going through the transformer container in raw deployment mode transformer collocation. This is due to the kubernetes service points to the predictor container port instead of transformer container port.

```yaml
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: custom-transformer-collocation
spec:
  predictor:
    containers:
      - name: kserve-container        # Do not change the name; This should be the predictor container
        image: "pytorch/torchserve:0.9.0-cpu"
        args:
          - "torchserve"
          - "--start"
          - "--model-store=/mnt/models/model-store"
          - "--ts-config=/mnt/models/config/config.properties"
        ports:
          - containerPort: 8085
            protocol: TCP
          
        env:
          - name: TS_SERVICE_ENVELOPE
            value: kserve
          - name: STORAGE_URI    # This will trigger storage initializer; Should be only present in predictor container
            value: "gs://kfserving-examples/models/torchserve/image_classifier/v1"
        resources:
          requests:
            cpu: 100m
            memory: 256Mi
          limits:
            cpu: 1
            memory: 1Gi

      - name: transformer-container    # Do not change the container name
        image: kserve/image-transformer:latest
        args:
          - --model_name=mnist
          - --protocol=v1    # protocol of the predictor; used for converting the input to specific protocol supported by the predictor
          - --http_port=8082
          - --grpc_port=8081
          - --predictor_host=localhost:8085      # predictor listening port
        ports:
          - containerPort: 8082
            protocol: TCP
            name: http
        resources:
          requests:
            cpu: 100m
            memory: 256Mi
          limits:
            cpu: 1
            memory: 1Gi
```

**What did you expect to happen:**
The requests should be go through the transformer.

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version:
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3317

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func setDefaultPodSpec(podSpec *corev1.PodSpec) {
 			container.ImagePullPolicy = corev1.PullIfNotPresent
 		}
 		// generate default readiness probe for model server container
-		if container.Name == constants.InferenceServiceContainerName {
+		if container.Name == constants.InferenceServiceContainerName || container.Name == constants.TransformerContainerName {
```
Comment (terrytangyuan) [inline pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go:145]: Update the comment above as well?

**Comment by israel-hdez** [comment]
I haven't explored Raw, so this may be a naive question but: Is it possible to use `spec.transformer` to get the expected outcome?

I find weird to pack everything into `spec.predictor`.

**Comment by sivanantha321** [comment]
> I haven't explored Raw, so this may be a naive question but: Is it possible to use `spec.transformer` to get the expected outcome?
> 
> I find weird to pack everything into `spec.predictor`.

We are discussing about that. Feel free to add your opinions on collocation of transformer [here](https://github.com/kserve/kserve/issues/3269). For custom models predictor and transformer containers are packed under `spec.predictor` for now. For more info see the [docs](https://kserve.github.io/website/latest/modelserving/v1beta1/transformer/collocation/)

File: pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go (hunk 1 above)
Comment (sivanantha321) [reply pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go:145]: done

**Pull request merged**